### ISSUE TITLE
Linux ready

### ISSUE THREAD

**Issue by TechnoWolfEver** [Zaarrg/TTVDropBot#26]
Hey,
would be a cool feature if you make an linux version for it,
easy to make (I guess?!) since many people got a little linux server to run a few things its cool to run the script there so you dont need your pc to be online.

with kind regards

**Comment by Reserton** [comment]
I have not tried it yet, but as it can uses npm, you should be able to just install node.js in linux and install the npm version.

**Comment by Zaarrg** [comment]
@TechnoWolfEver Compiled the bot for linux-x64. Dont have any pc to test it on so i dont know if its gonna work. Would be nice if you can check it. Updated the [1.3.3.1 Release](https://github.com/Zaarrg/TTVDropBot/releases/tag/1.3.3.1).

**Comment by Reserton** [comment]
I tried to run that file in linux and I think my linux config did not support it.

Currently trying npm and that appears like it will work once I solve the issue I have.

I use Linux headless without a GUI. However the intial browser window opens non headless. Does this require to run with the gui or can the intial window also run as headless without issue?

**Comment by Reserton** [comment]
I got the npm version working fine in Linux, but it just refuses to work with profile directory, it always asks me to login, so I cannot do it with a headless (no gui) lkinux.

**Comment by TechnoWolfEver** [comment]
so today i woke up seeing whats happening here,
thanks for the help and afford.
so i installed npm, node and so on and tried starting it i dont get it starting since i'm opening it on a guiless Linux server.

![image](https://user-images.githubusercontent.com/46536097/126112502-d3b0e015-46c7-4a58-811b-bf9dc0cc4202.png)

**Comment by Reserton** [comment]
Yeah, Gui less does not seem to be possible. Especially sicne the user profile does not want to work.

**Comment by TechnoWolfEver** [comment]
> Yeah, Gui less does not seem to be possible. Especially sicne the user profile does not want to work.

yea servers dont got any browsers installed guess thats the problem.

**Comment by Reserton** [comment]
I installed the browser, but issue is it cant pull the profile directory info so you have to login which you cant do without a gui.

Even if the profile directory worked however the video then is refusing to play in a virtual machine for me which stops the progress going up.

**Comment by Reserton** [comment]
After more testing I think the user profile not working is an issue with the program not chromium or Linux.

If I do following command it works fine now and is logged into Twitch : 

chromium-browser --user-data-dir="/home/rustdrops/chomeprofile/" http://www.twitch.tv

However when I add same user data directory into TTVDropBot it just opens Twitch login everytime.

Is it possible the TTVDropBot does some form of check on the profile directory and is determining the Linux path is incorrect and forcing login?

**Comment by BlazTrampus** [comment]
i keep getting this error while trying to run the bot on a raspberry pi
![1](https://user-images.githubusercontent.com/59237118/133759112-a312096b-72a9-42c2-a3d8-27da4dd01832.png)
anyone have any idea what could be the cause of this?

**Comment by Zaarrg** [comment]
Not Active anymore, reopen if error still occurs
